### PR TITLE
Documenter Version of Docs

### LINKED ISSUE #11: Move website to use Documenter.jl
The current website is completely in HTML and not particularly fun to edit. I'd much prefer to move to use something like Documenter.jl, like what we did for JuliaPhysics website: https://github.com/JuliaPhysics/juliaphysics.github.io

The layout will probably need to change (the current tables are probably not easy to reproduce with Documenter.jl), but I think that the easy of use is worth the change.

### PR DESCRIPTION
**Pull request by mileslucas**
Looks like this: http://docs-temp-dustextinction.s3-website-us-west-1.amazonaws.com/index.html

Fix #11

### CODE REVIEW

**Comment by giordano** [comment]
I think that the target branch should be different from `master`, because for github.io pages the `master` branch is reserved to the published website.  For example, for [`juliaphysics.github.io`](https://github.com/JuliaPhysics/juliaphysics.github.io) the branch is called `source`

**Comment by mileslucas** [comment]
good point

File: make.jl [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+makedocs(sitename = "JuliaAstro",
+    format = Documenter.HTML(prettyurls = get(ENV, "CI", nothing) == "true",
+        assets = [
+            "assets/favicon.ico",
+        ],
+    ),
+    strict = true
+)
+
```
Comment (giordano) [inline make.jl]: See the [`make.jl` script](https://github.com/JuliaPhysics/juliaphysics.github.io/blob/3e01bd7c43fbf43ca00414b784c06e44c126ee8d/make.jl) of `juliaphysics.github.io`, it has also the `deploydocs` function, to indicate which is the source branch (`source`), and which is branch where the pages will be saved to (`master`)

File: src/index.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+Use the packages, file issues, and join the conversation!
+
+- [JuliaAstro](https://github.com/juliaastro) on GitHub
+- [julia-astro](https://groups.google.com/forum/#!forum/julia-astro) mailing list on Google Groups
+- [Astro/Space](https://discourse.julialang.org/c/domain/astro) topics on JuliaLang Discourse
+- [#JuliaAstro:openastronomy.org](https://riot.im/app/#/room/#JuliaAstro:openastronomy.org) on Matrix
+- [#astronomy](slack://channel?id=CMXU6SD7V&team=T68168MUP) on [JuliaLang Slack](https://slackinvite.julialang.org/)
+
+## Google Summer of Code
+Would you like to contribute to JuliaAstro as part of Google Summer of Code? You can! We are member of OpenAstronomy, go to the GSoC page to see how to apply and the list of ideas. You can also propose us your own idea!
```
Comment (giordano) [inline src/index.md]: A link to [OpenAstronomy](https://openastronomy.org/)?

**Comment by giordano** [comment]
> do you mind pushing a dummy branch to source?

Done!

**Comment by mileslucas** [comment]
only thing I can't test is CI and deployment. Builds fine locally, though.

**Comment by giordano** [comment]
I never published pages with GitHub actions, do we need to setup a token like when using Travis?

**Comment by mileslucas** [comment]
yeah, let me slack message you the secret (can go straight into `DOCUMENTER_KEY` in https://github.com/juliaastro/juliasastro.github.io/secrets

**Pull request merged**